mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -534,8 +577,15 @@ class Word2VecModel private[spark] (
     // Need not divide with the norm of the given vector since it is constant.
     val cosVec = cosineVec.map(_.toDouble)
     var ind = 0
+    var vecNorm = 1f
+    if (norm) {
```

Comment:
I think we should make it return proper cosine similarity. @jkbradley thoughts? 

Note that there is an outstanding JIRA SPARK-7617 and PR #6245 that addresses the normalization, as well as caching the normalized vectors. Though it is a bit stale now - if @ezli will be updating that PR soon we can make those changes there, otherwise leave them for a follow-up PR?